verification/curator-service/ui/src/components/Users.test.tsx

```diff
@@ -110,3 +112,117 @@ test('updates roles on selection', async () => {
     expect(await findByText('reader')).toBeInTheDocument();
     expect(await findByText('curator')).toBeInTheDocument();
 });
+
+test('calls callback when user is changed', async () => {
+    let functionCalledCounter = 0;
+    const user =
+    {
+        _id: 'abc123',
+        name: 'Alice Smith',
+        email: 'foo@bar.com',
+        roles: ['admin', 'reader'],
+    };
+    const axiosResponse = {
+        data: {
+            users: [user],
+            total: 1,
+        },
+        status: 200,
+        statusText: 'OK',
+        config: {},
+        headers: {},
+    };
+    mockedAxios.get.mockResolvedValueOnce(axiosResponse);
+
+    const { findByText, getByRole } = render(
+        <Users user={user} onUserChange={() => { functionCalledCounter++; }} />,
+    );
+    expect(await findByText('Alice Smith')).toBeInTheDocument();
+
+    const updatedUser =
+    {
+        _id: 'abc123',
+        name: 'Alice Smith',
+        email: 'foo@bar.com',
+        roles: ['admin', 'reader', 'curator'],
+    };
+    const axiosPutResponse = {
+        data: {
+            users: [updatedUser],
+        },
+        status: 200,
+        statusText: 'OK',
+        config: {},
+        headers: {},
+    };
+    mockedAxios.put.mockResolvedValueOnce(axiosPutResponse);
+    expect(functionCalledCounter).toBe(0);
+
+    // Select new role
+    fireEvent.mouseDown(getByRole('button'));
```

Comment:
I'm thinking of this now but will all these tests break once we add another button to this page? Can we be a bit more explicit that this is a button to change the roles?

Reply:
Took some ugliness to get a test id in the right place but finally achieved! Unfortunately I couldn't find a way of chaining lookups e.g. execute getByText() on only children of a different getByTestId() call :/